core/src/main/java/org/apache/gravitino/utils/PrincipalUtils.java

```diff
@@ -44,13 +44,19 @@ public static <T> T doAs(Principal principal, PrivilegedExceptionAction<T> actio
     }
   }
 
+  // This method can't be used in nested `Subject#doAs` block.
   public static Principal getCurrentPrincipal() {
     java.security.AccessControlContext context = java.security.AccessController.getContext();
     Subject subject = Subject.getSubject(context);
-    if (subject == null || subject.getPrincipals(UserPrincipal.class).isEmpty()) {
```

Comment:
Why do you change this？

Reply:
I've modified it according to my coding habits, if you feel is not okay, I will revert it.